src/components/furniture/Shelf.vue

```diff
@@ -1,120 +1,63 @@
 <template>
-  <div class="flex-container">
-    <div class="shelf-height" :style="cssStyle" @click="openUpdateModal">
-      <div v-if="isUpdating">
-        <div class="update-input">
-          <input
-            type="number"
-            v-model="newHeight"
-            :placeholder="shelf.height"
-          /><span>cm</span>
-        </div>
-      </div>
-      <p v-else>{{ shelf.height }} cm</p>
-    </div>
-    <button v-if="isUpdating" class="button" @click="updateFurniture">
-      OK
-    </button>
+  <div v-if="logic === 'shelfHeightLogic'">
+    <shelf-height-logic
+      :myShelf="myShelf"
+      :style="cssStyle"
+      class="shelf"
+    ></shelf-height-logic>
+  </div>
+  <div v-if="logic === 'addSeparatorsLogic'">
+    <add-separators-logic
+      :myShelf="myShelf"
+      :style="cssStyle"
+      @updating-separators="emitUpdatingSeparators"
+      class="shelf"
+    ></add-separators-logic>
+  </div>
+  <div v-if="logic === 'addHardware'" :style="cssStyle" class="shelf">
+    <space
+      v-for="space in myShelf.insideSpaces.spaces"
+      :key="space.id"
+      :mySpace="space"
+      :myShelf="myShelf"
+      logic="addHardware"
```

Comment:
We can go even further and make the component absolutely ignorant of any context.

We just tell him true or false: 

```suggestion
      :addHardware="addHardware === 'addHardware'"
```
so this will return a true or false and thus creating a boolean prop.

Reply:
by the way, it's weird that a variable `addHardware` would be a string with the name of the variable inside the string `'addHardware'`. I smell something here but I let you think about it.